src/components/create-exam/publish-button.tsx

```
@@ -18,77 +21,194 @@ import { Modal } from "@/components/ui/modal"; // Our friendly, custom modal
+
+          if (!("mina_signer_payload" in deployTx)) {
+            toast.error("Failed to create exam. Could not build deploy transaction");
+            setIsPublishing(false);
+            return;
+          }
+
+          const { mina_signer_payload, serializedTransaction, contractAddress } = deployTx;
+
+          const signedAuroData = window.mina?.isPallad
+            ? ((
+                await window?.mina?.request({
```

Biome (1.9.4)

[error] Unsafe usage of optional chaining.

If it short-circuits with 'undefined' the evaluation will throw TypeError here:


(lint/correctness/noUnsafeOptionalChaining)